### PR TITLE
Entity input handling 

### PR DESCRIPTION
**Pull request by martyworm**
Entities handle input like cards do now.

Added better looking mana.

Turns are working great now, you can only hover on opposite turn but cannot interact or make it do anything. Players are now gaining 3 mana per turn, which is capped at 10 for now, and the minions when cast have 0 action points but gain points per turn based on their actionPointsPerTurn stat.

Fixed the movement bug where you could move one space, then go back a space and your AP would go back up. However, I don't know how stable it is. I'm using pixelsMoved and incrementing it during moveX and moveY.

Can no longer cast minions from active, and can only cast from hand.

better gui overlay based on whose turn it is

### CODE REVIEW

File: src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java [hunk 1]
```diff
@@ -22,16 +22,16 @@ public CardSkeleton(Handler handler, int id, int playerNumber){
...
+        this.actionPointsPerTurn = 4;
         this.spells = 1;
         this.health = 20;
     }
 
     @Override
     public void cast(Tile t){
-
-        if(manaCheck(handler.getBattle().getCurrentPlayer())) {
+        if(doesPlayerHaveEnoughMana(handler.getBattle().getCurrentPlayer())) {
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java]: naming - canCastSpell

File: src/main/java/com/martyworm/Battle/Battle.java [hunk 2]
```diff
@@ -93,28 +88,24 @@ public void onClick() {
 
     public void onMouseMove(MouseEvent e){
         tileManager.onMouseMove(e);
-        entityManager.onMouseMove(e);
-        player1.getCardManager().onMouseMove(e);
-        player2.getCardManager().onMouseMove(e);
+        player1.onMouseMove(e);
```
[the remark this answers is not in the document]
Comment (martyworm) [reply src/main/java/com/martyworm/Battle/Battle.java]: I figure you should still be able to hover and see info on enemy minions even when it isn't your turn.

File: src/main/java/com/martyworm/Battle/Battle.java [hunk 3]
```diff
@@ -159,6 +148,11 @@ public Player getCurrentPlayer(){
         else return player2;
```
[the remark this answers is not in the document]
Comment (martyworm) [reply src/main/java/com/martyworm/Battle/Battle.java]: As far as the game knows there's 10 players.  Why would it default to player2 without the else?

File: src/main/java/com/martyworm/Battle/Battle.java [hunk 4]
```diff
@@ -173,20 +167,14 @@ public void setHandler(Handler handler) {
     }
 
     public EntityManager getEntityManager() {
-        return entityManager;
+
+        return getCurrentPlayer().getEntityManager();
```
[the remark this answers is not in the document]
Comment (martyworm) [reply src/main/java/com/martyworm/Battle/Battle.java]: As mentioned in that text the other day, I changed entity managers to be owned by the players, just like the card managers are.  Each player with their own entities and cards.  And input handling is easier this way.

File: src/main/java/com/martyworm/Player/Player.java [hunk 5]
```diff
@@ -37,34 +38,63 @@ public Player(Handler handler, int id){
...
         if(id == 1) {
-            for (int i = 0; i < 6; i++) {
+            for (int i = 0; i < 12; i++) {
                 if (mana == i) {
                     g.drawImage(Assets.manaBubble[i], PLAYER1_MANA_DISPLAY_SPOT_X, PLAYER1_MANA_DISPLAY_SPOT_Y, null);
                 }
             }
         }else if (id == 2){
-            for (int i = 0; i < 6; i++) {
+            for (int i = 0; i < 12; i++) {
```
[the remark this answers is not in the document]
Comment (martyworm) [reply src/main/java/com/martyworm/Player/Player.java]: Roger that

**Pull request merged**